addons/sourcemod/scripting/szf/pickupweapons.sp

```diff
@@ -77,7 +78,18 @@ public Action Event_WeaponsRoundStart(Event event, const char[] name, bool dontB
 				//If rare weapon cap is unreached, make it a "rare" weapon
 				if (iRare > 0)
 				{
-					SetRandomWeapon(iEntity, WeaponRarity_Rare);
+					Weapon wep;
+					
+					//If array is null or empty, fill it with the weapon list
+					if (g_aRareWeapons == null || g_aRareWeapons.Length == 0)
+					g_aRareWeapons = GetAllWeaponsWithRarity(WeaponRarity_Rare);
```

Comment:
`g_aRareWeapons` need to be `delete`d before creating new array, as `g_aRareWeapons.Length == 0` still counts as a handle giving memory leak.